terra-app-setup-chart/templates/service-account.yaml

```diff
@@ -5,4 +5,6 @@ metadata:
   labels:
     leonardo: "true"
   annotations:
-    iam.gke.io/gcp-service-account: {{ .Values.serviceAccount.annotations.gcpServiceAccount | quote }}
\ No newline at end of file
+    {{- if eq .Values.cloud "gcp" -}}
```

Comment:
Will likely add Workload Identity annotations here as part of https://broadworkbench.atlassian.net/browse/TOAZ-228